server/api/impl/store/manifest.go

```diff
@@ -0,0 +1,266 @@
+// Copyright (c) 2020-present Mattermost, Inc. All Rights Reserved.
+// See License for license information.
+
+package store
+
+import (
+	"crypto/sha1" // nolint:gosec
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	"github.com/pkg/errors"
+
+	"github.com/mattermost/mattermost-plugin-apps/apps"
+	"github.com/mattermost/mattermost-plugin-apps/awsclient"
+	"github.com/mattermost/mattermost-plugin-apps/server/api"
+	"github.com/mattermost/mattermost-plugin-apps/server/utils"
+	"github.com/mattermost/mattermost-plugin-apps/server/utils/httputils"
+)
+
+// manifestStore combines global (aka marketplace) manifests, and locally
+// installed ones. The global list is loaded on startup. The local manifests are
+// stored in KV store, and the list of their keys is stored in the config, as a
+// map of AppID->sha1(manifest).
+type manifestStore struct {
+	*Store
+
+	mutex sync.RWMutex
+
+	global map[apps.AppID]*apps.Manifest
+	local  map[apps.AppID]*apps.Manifest
+}
+
+var _ api.ManifestStore = (*manifestStore)(nil)
+
+func (s *manifestStore) InitGlobal(awscli awsclient.Client, bucket string) error {
+	bundlePath, err := s.mm.System.GetBundlePath()
+	if err != nil {
+		return errors.Wrap(err, "can't get bundle path")
+	}
+	assetPath := filepath.Join(bundlePath, "assets")
+	f, err := os.Open(filepath.Join(assetPath, api.ManifestsFile))
+	if err != nil {
+		return errors.Wrap(err, "failed to load global list of available apps")
+	}
+	defer f.Close()
+
+	return s.initGlobal(awscli, bucket, f, assetPath)
+}
+
+// initGlobal reads in the list of known (i.e. marketplace listed) app
```

Comment:
would be good to add a comment as to why (unlike apps) we do not store global manifests in the KV store.